src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -153,18 +161,51 @@ const peerConnectRequestSignChangeHandler = async (
   );
 };
 
+const peerConnectedChangeHandler = async (
+  event: PeerConnectedEvent,
+  dispatch: ReturnType<typeof useAppDispatch>
+) => {
+  const peerConnection =
+    await Agent.agent.peerConnectionMetadataStorage.getPeerConnectionMetadata(
+      event.payload.dAppAddress
+    );
+  const existingConnections =
+    await Agent.agent.peerConnectionMetadataStorage.getAllPeerConnectionMetadata();
+  dispatch(setWalletConnectionsCache(existingConnections));
+  dispatch(setConnectedWallet(peerConnection));
```

Comment:
hmm, do we need the entire peer connection in Redux for connected wallet? I think we can just have `event.payload.dAppAddress` so its more lightweight